### ISSUE TITLE
Version 4.x no longer compiling to es5

### ISSUE THREAD

**Issue by walkjivefly** [MetaMask/metamask-extension#3505]
Expected behaviour: not a blank window. I don't know what it should look like, never used it before.
Actual behaviour: blank window.

![metamask_01a](https://user-images.githubusercontent.com/17745270/37237738-ea06caba-23e5-11e8-805d-d3590221a4d0.png)

Metamask version: 4.2.0
Browser: Chrome Version 46.0.2490.71  (yes, very old, but it's the latest version I can run on this particular machine).
OS: Linux SolydK 201412 (bastardised Debian Wheezy/Jessie hybrid)

Popup log says:

![metamask_01b](https://user-images.githubusercontent.com/17745270/37237747-0ce0daee-23e6-11e8-9bb3-96015fa51f2b.png)

And background log says:

![metamask_02a](https://user-images.githubusercontent.com/17745270/37237748-159a026e-23e6-11e8-9fcc-6b8b642ef1a9.png)

These logs don't match any of the other "blank window" reports.

**Comment by lazaridiscom** [comment]
> Browser: Chrome Version 46.0.2490.71 (yes, very old, but it's the latest version I can run on this particular machine).

current version: Version 64.0.3282.186 (Official Build) (64-bit)

Your version is "Tuesday, October 13, 2015", nearly 2.5 years old.

The whole Ethereum Domain is a very new and moving technology. So, going with an 2,5 years old browser is most possibly not the way to go. You can await the response of the team.

@danfinlay , besisdes #3343, MM should verify its "minimum requirements" on startup.

**Comment by walkjivefly** [comment]
Chrome web store says it's compatible.

**Comment by kumavis** [comment]
@walkjivefly as @lazaridiscom said your browser is **too old**. At 2.5 years, its not just a matter of not having the latest features, I'm not confident that your browser is safe to use. If a critical vulnerability is known that affects your browser version, websites might be able to steal your private keys from MetaMask.

That said, you mentioned it used to work. Its likely that with version 4.x we changed our build process and are no longer compiling es7 to es5.

**Comment by kumavis** [comment]
For now, you can manually install an older version.
- Be aware it will not auto update.
- Be sure you have your seed phrase written down on paper before starting this process.
- Download a version from before 4.x (https://github.com/MetaMask/metamask-extension/releases?after=v4.1.0)
- You can drag and drop the downloaded version in the extensions page of chrome.
- Be sure to disable the old version of metamask, as things break if two are open at once.

**Comment by walkjivefly** [comment]
Thanks for the suggestion. I tried 3.14.1 and all 3.x.0 back to 3.4.0; none of them worked. I've never used Metamask before; I think someone else reported they had in some other issue with similar symptoms. I only wanted to install it to have a look at the Origin Protocol demo dApp. Guess I won't be bothering for the time being.

Changing the title won't help other people in the same situation as me find the report and see the suggested workaround. Which is a shame since it might work for them.

If I ever get my machine upgraded I'll have another go. Cheers!

**Comment by danfinlay** [comment]
> Changing the title won't help other people in the same situation as me find the report and see the suggested workaround.

But it will help us fix the issue sooner. Sorry for the hassle, have a good one!

**Comment by ryan-rowland** [comment]
Looks like this was introduced here https://github.com/MetaMask/metamask-extension/pull/2975/files#diff-b9e12334e9eafd8341a6107dd98510c9L381

**Comment by kumavis** [comment]
The "fix" was reverted because it broke the build process

**Comment by ryan-rowland** [comment]
🤦‍♂️ Sorry I broke your build.

Been looking at this for way too long now. Things I found:
1) The "memory leak" removing the babel step was just a memory overflow because uglify was trying to package your whole library at once because it was all in the same file. Running node with `node --max_old_space_size=4096` fixed this.
2) The "memory leak" was also covering up another issue: running babel wraps the code in a `require` block, which also breaks everything once loaded into the browser (same breaking behavior my commit caused)
3) Babel-runtime is huge and almost doubling the size of some minified files
4) Removing `const` declarations with babel/browserify is makes me want to pull my hair out.

Tapping out. 🤷‍♂️

**Comment by ryan-rowland** [comment]
Also: CircleCI approved my PR despite it breaking the build, we might want to update the CI to cover this step of the build process.

**Comment by danfinlay** [comment]
Captured in #3699.

**Comment by Epelzhenya** [comment]
Can anybody tell me with simple language what to do whith this problem? English is not my native language.

**Comment by lazaridiscom** [comment]
@Epelzhenya, just update to a newer version of your browser.

**Comment by Epelzhenya** [comment]
@lazaridiscom thank you wery much)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.2 ETH (123.94 USD @ $619.68/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $14,740.79 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 4 weeks from now. Please review their questions below:


1. [JonnyBurger](https://gitcoin.co/profile/JonnyBurger) has started work. 	
 * Q: I will experiment with different babel presets, like babel-preset-es2015 and hope to find a solution that doesn't break the build and keeps the size small.

**Comment by gitcoinbot** [comment]
@JonnyBurger Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=1) | [3 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=3) | [5 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=5) | [10 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=10) | [100 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 3 weeks, 4 days from now. Please review their questions below:


1. [Platinumwrist](https://gitcoin.co/profile/Platinumwrist) has started work. 	
 * Q: Help me understand this issue furthermore thanks

**Comment by JonnyBurger** [comment]
Hi there,

sorry I have to give up this as well. 
I figured out the memory issue at least.

The build step was bundle -> minify -> transpile, which gave one big file to Babel and that is why the memory issue occured.
Changing it to transpiling -> bundling -> minifying solved this issue.

However, it's true that is seems very hard to configure Babel to remove all `const`. Alternatively, I was also not able to set "strict mode" in Babel. Weird because it should work.

So sorry for not being able to deliver.

**Comment by vs77bb** [comment]
Thanks for the honest update @JonnyBurger and appreciate you taking a look! Have a great weekend.

**Comment by gitcoinbot** [comment]
@Platinumwrist Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=1) | [3 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=3) | [5 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=5) | [10 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=10) | [100 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=100)

**Comment by gitcoinbot** [comment]
@Platinumwrist Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=1) | [3 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=3) | [5 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=5) | [10 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=10) | [100 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=100)

**Comment by gitcoinbot** [comment]
@Platinumwrist Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=1) | [3 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=3) | [5 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=5) | [10 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=10) | [100 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=100)

**Comment by erchaves** [comment]
I'm not 100% sure, but at a glance, I think Babel is not setup to transpile const and let, but [traceur](https://github.com/google/traceur-compiler/wiki/Getting-Started) could do this for you. It doesn't seem like it would make sense to introduce that into the actual codebase to solve this problem just for really old legacy browsers. But if some folks _really_ needed to find a work around, I suppose they could filter the js through traceur and bootstrap it back in, maybe through a fork of the project or something..

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 2 weeks from now. Please review their questions below:


1. [rafaelboyero](https://gitcoin.co/profile/rafaelboyero) has started work.

**Comment by gitcoinbot** [comment]
@rafaelboyero Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=1) | [3 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=3) | [5 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=5) | [10 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=10) | [100 days](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555?snooze=100)

**Comment by WillBeebe** [comment]
Babel can transpile const and let for sure, you just need to set the right target. It will even add `"use strict"` to the top of the file.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 week, 4 days ago. Please review their questions below:


1. [ajnauleau](https://gitcoin.co/profile/ajnauleau) has started work. 	
 * Q: Utilize Babel for JavaScript compatibility and transpilation
- JavaScript package that transpiles JavaScript ES6+ code to ES5

Install babel dependencies by running

npm install babel-cli -D
npm install babel-preset-env -D

Create a .babelrc file inside your project and add the following code inside it:

{
  "presets": ["env"]
}

Add the following script to your scripts object in package.json:

"build": "babel src -d lib"

**Comment by ajnauleau** [comment]
Looking to be a bit more difficult than I expected, getting errors building gulp at mascara with babel "env"-- will keep posted.

**Comment by ajnauleau** [comment]
*Fixed, updated to babel-preset-env, just had to change the order of presets to "env", "react", "stage-0"-- now builds.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 0.2 ETH (95.15 USD @ $475.76/ETH) has been submitted by__:

 1. [@ajnauleau](https://gitcoin.co/profile/ajnauleau)



@vs77bb please take a look at the submitted work:
* [PR](https://github.com/MetaMask/metamask-extension/pull/4812) by @ajnauleau

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
$28,054.10 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 0.2 ETH (53.15 USD @ $265.77/ETH)  attached to this issue has been approved & issued to @ajnauleau.__  

 * Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MetaMask/metamask-extension/3505/555)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $17,078.38 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by lazaridiscom** [comment]
@bdresser , it looks like this one can be closed.
